tests/benchmark/test_baseline_benchmarks.py

```diff
@@ -0,0 +1,322 @@
+""" Profiles basic -jX functionality """
+# Copyright (c) 2020 Frank Harrison <doublethefish@gmail.com>
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+# pylint: disable=protected-access,missing-function-docstring,no-self-use
+
+import os
+import pprint
+import time
+
+import pytest
+
+from pylint.checkers.base_checker import BaseChecker
+import pylint.interfaces
+from pylint.lint import check_parallel, PyLinter, Run
+from pylint.utils import register_plugins
+from pylint.testutils import TestReporter as Reporter
+
+
+def _empty_filepath():
+    return os.path.abspath(
+        os.path.join(
+            os.path.dirname(__file__), "..", "input", "benchmark_minimal_file.py"
+        )
+    )
+
+
+class SleepingChecker(BaseChecker):
+    """ A checker that sleeps, the wall-clock time should reduce as we add workers
+
+    As we apply a roughly constant amount of "work" in this checker any variance is
+    likely to be caused by the pylint system. """
+
+    __implements__ = (pylint.interfaces.IRawChecker,)
+
+    name = "sleeper"
+    msgs = {"R9999": ("Test", "test-check", "Some helpful text.",)}
+    sleep_duration = 0.5  # the time to pretend we're doing work for
+
+    def process_module(self, _astroid):
+        """ Sleeps for `sleep_duration` on each call
+
+        This effectively means each file costs ~`sleep_duration`+framework overhead """
+        time.sleep(self.sleep_duration)
+
+
+class SleepingCheckerLong(BaseChecker):
```

Comment:
It look like it's the same or almost the same than the `SleepingChecker`. I suppose the sleep duration will be higher in `SleepingCheckerLong` later ? Maybe, we could inherit from a `BaseSleepingChecker` and just change some attribute to reduce duplication.

Reply:
Ha my bad, it's a test, it's supposed to be like this. 😅